src/ng/directive/attrs.js

```diff
@@ -341,22 +341,35 @@
 
 var ngAttributeAliasDirectives = {};
 
-
 // boolean attrs are evaluated
 forEach(BOOLEAN_ATTR, function(propName, attrName) {
   // binding to multiple is not supported
   if (propName == "multiple") return;
 
-  var normalized = directiveNormalize('ng-' + attrName);
+  function defaultLinkFn(scope, element, attr) {
+    scope.$watch(attr[normalized], function ngBooleanAttrWatchAction(value) {
+      attr.$set(attrName, !!value);
+    });
+  }
+
+  var normalized = directiveNormalize('ng-' + attrName),
+    linkFn = defaultLinkFn;
+
+  if (propName === 'checked') {
+    linkFn = function(scope, element, attr) {
+      // ensuring ngChecked doesn't interfere with ngModel when both are set on the same input
+      // - https://github.com/angular/angular.js/issues/10662
+      if (attr.ngModel !== attr[normalized) {
```

Comment:
kinda feel like special casing different attributes in here is kind of gross, too --- but we can worry about that later